src/rpi.py

```diff
@@ -0,0 +1,132 @@
+import binascii
+import logging
+import subprocess
+import os
+import shutil
+import tempfile
+import hashlib
+from datetime import datetime
+
+from utils import RPI_DIR
+
+LOG = logging.getLogger("rpi")
+
+
+def keys():
+    dirs = os.listdir(RPI_DIR)
+    response = {
+        "keys": [dirs],
+    }
+    return response
+
+
+def new(body, user):
+    tmpdir = tempfile.mkdtemp(prefix="rpi-")
+    cmd = [ "openssl", "genpkey", "-algorithm", "RSA", "-out",
+           tmpdir + os.sep + "private.pem", "-pkeyopt",
+           "rsa_keygen_bits:2048" ]
+    cmd_result = subprocess.run(cmd, stdout=subprocess.DEVNULL,
+                                stderr=subprocess.DEVNULL, check = False)
+    if cmd_result.returncode != 0:
+        raise RuntimeError("Failed to generate RPI key pair")
+
+    cmd = [ "openssl", "rsa", "-pubout", "-in", tmpdir + os.sep + "private.pem", "-out",
+           tmpdir + os.sep + "public.pem" ]
+    cmd_result = subprocess.run(cmd, stdout=subprocess.DEVNULL,
+                                stderr=subprocess.DEVNULL, check = False)
+    if cmd_result.returncode != 0:
+        raise RuntimeError("Failed to generate RPI key pair")
+
+    cmd = [ "ssh-keygen", "-l", "-f" ,  tmpdir + os.sep + "private.pem" ]
+    cmd_result = subprocess.run(cmd, stdout=subprocess.PIPE, check = False)
+    if cmd_result.returncode != 0:
+        raise RuntimeError("Failed to generate RPI key pair")
+
+    fingerprint = hashlib.md5(cmd_result.stdout.split()[1]).hexdigest()
+    shutil.move(tmpdir, RPI_DIR + os.sep + fingerprint)
+    response = {"fingerprint": fingerprint}
+
+    LOG.info("%s successfully generated a new RPI key %s", user, fingerprint)
+
+    return response
+
+
+def key(key_id):
+    for entry in os.listdir(RPI_DIR):
```

Comment:
Sounds like an overkill to always iterate over all the keys, maybe just try to open the file and return 404 if it fails?